Assessment-1/Services/RiderService.cs

```diff
@@ -0,0 +1,66 @@
+﻿using Assessment_1.Entities;
+using Assessment_1.Enums;
+using Assessment_1.Exceptions;
+using Assessment_1.Mappers;
+using Assessment_1.Models;
+using Assessment_1.Services.IServices;
+using Assessment_1.Utils;
+
+namespace Assessment_1.Services
+{
+    public class RiderService : IRiderService
+    {
+        private IUserService _userService;
+        private IDriverService _driverService;
+        private ITripService _tripService;
+
+        public RiderService(IUserService userService, IDriverService driverService, ITripService tripService)
+        {
+            _userService = userService;
+            _driverService = driverService;
+            _tripService = tripService;
+        }
+
+        public TripRequestResponse RequestTrip(TripRequest tripRequest, string userEmail, UserRole userRole)
+        {
+            User rider = _userService.GetUserFromEmailAndRole(userEmail, userRole);
+            Guid riderId = rider.UserId;
+
+            try
+            {
+                Trip ongoingTrip = _tripService.GetCurrentRiderTrip(riderId);
+                throw new TripOperationException("The rider is already in an ongoing trip.");
+            }
+            catch (TripNotFoundException)
+            {
+                Driver availableDriver = _driverService.FindAvailableDriver(tripRequest.RideType);
+                Guid driverId = availableDriver.UserId;
+
+                _driverService.SetDriverAvailability(availableDriver.DriverId, DriverAvailability.InARide);
+
+                string otp = OtpGenerator.GenerateOtp();
+
+                Trip trip = TripMapper.CreateTripFromTripRequest(tripRequest, riderId, driverId, otp);
+
+                _tripService.AddTrip(trip);
+
+                return TripMapper.CreateTripRequestResponseFromTripAndDriver(trip, availableDriver, otp);
+            }
+        }
```

Comment:
INstead of throwing an exception in FindAvailableDrive()

throw in the upper layer so that here you don't have to catch exception for a positive case.